src/api/walk.js

```diff
@@ -177,12 +177,15 @@ import { join } from '../utils/join.js'
  *
  * Example 2: Return the difference between the working directory and the HEAD commit
  * ```js
- * const diff = require('diff-lines')
- * async function map(filepath, [head, workdir]) {
+ * import diff from 'diff-lines'
```

Comment:
I think we need to stay consistent in the docs whether we're using ESM or CommonJS. Thus for, we have used CommonJS, which is a common denominator. I think we should stick with it here.

Reply:
CommonJs is a niche tech in the grand scheme of things. of the 36M+ JavaScript developers, less than 20% use Node. Of those who use node, ~50%+ strongly prefer typescript/esm.  Calling a legacy tech on its way out used by less than 5% of the user base, the LCD, is miopic.

I understand where you're coming from in the package maintenance perspective, as it's usually not wise to publish an esm only package at the moment... unless it's a plugin for an esm centric build system like Webpack 5, ESBuild or Vite.

Reply:
If you really think this way please rewrite the whole documentation to use ES Modules. If you don't want to do that then please use whatever is in the project and not introduce a new style or syntax in a random place. If you want to contribute you need to follow the style of the project.

Reply:
This is the first rule of Open Source. Never introduce a new style only follow whatever is in the project. Or your PR will be rejected.

Reply:
> This is the first rule of Open Source. Never introduce a new style only follow whatever is in the project. Or your PR will be rejected.

This is not a style change this is an English documentation improvement. Documentation is not code and can only be slowly improved upon. If you used tabs for indentation I would have submitted tabs for indentation, despite me being vehemently opposed to them.

Sidenote, the lack of an ESM export for the http module introduces a portability problem, and is a minor hurdle...

...fine I'll get rid of anything that doesn't work with Node v0.10

Reply:
You don't need to be sarcastic. NOTE that by default `import` doesn't work even in NodeJS 18. You need to create package.json file and use `type: module` to be able to use `import` or use some kind of feature flag if there is any.

Reply:
The require statement for the diff package is now missing. Can this be corrected?

Reply:
> 

Yes, in node that is the case. In browsers, deno and bun import works but require doesn't. In node also works if you use the mjs extension. No flags are needed in Node 16 LTS for ES Modules or top level await support.

Reply:
My bad. I didn't noticed that the code is right now broken. Because you use `diff` in the code that is not defined.